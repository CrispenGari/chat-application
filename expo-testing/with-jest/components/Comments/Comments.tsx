import { Text } from "react-native";
import React from "react";

const Comments = () => {
  const [data, setData] = React.useState<any>(undefined);
  React.useEffect(() => {
    fetch("https://cat-fact.herokuapp.com/facts")
      .then((response) => response.json())
      .then((data) => {
        setData(data);
      });
  }, []);

  if (data === undefined) {
    return <Text>Loading...</Text>;
  }
  return data.length > 0 ? (
    <Text>{data[0].text}</Text>
  ) : (
    <Text>No Comments!</Text>
  );
};

export default Comments;
