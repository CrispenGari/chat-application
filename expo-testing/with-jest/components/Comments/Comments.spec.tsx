import { render, screen, waitFor } from "@testing-library/react-native";
import Comments from "./Comments";

describe("<Comments/>", () => {
  it("renders correctly", () => {
    render(<Comments />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("displays comment", async () => {
    // @ts-ignore
    fetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve([
            { text: "first comment" },
            { text: "second comment" },
          ]),
      })
    );
    const { getByText } = render(<Comments />);
    await waitFor(() => {
      expect(fetch).toHaveBeenCalled();
    });
    const text = getByText("first comment");
    expect(text).toBeOnTheScreen();
  });

  it("displays no comment", async () => {
    // @ts-ignore
    fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve([]),
      })
    );

    const { getByText } = render(<Comments />);
    await waitFor(() => {
      expect(fetch).toHaveBeenCalled();
    });

    const catFactElement = getByText("No Comments!");
    expect(catFactElement).toBeOnTheScreen();
  });

  it("displays no comments due to loading", async () => {
    // @ts-ignore
    fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve([]),
      })
    );
    const { getByText } = render(<Comments />);
    const catFactElement = getByText("Loading...");
    expect(catFactElement).toBeOnTheScreen();
  });
});
