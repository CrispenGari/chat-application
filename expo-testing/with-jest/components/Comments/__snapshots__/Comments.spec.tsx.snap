// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments/> renders correctly 1`] = `
<Text>
  Loading...
</Text>
`;

exports[`<Comments/> renders correctly with initial value of 10 1`] = `
<Text>
  Loading...
</Text>
`;
